sdks/python/apache_beam/examples/dataframe/flight_delays.py

```diff
@@ -0,0 +1,129 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A word-counting workflow using dataframes."""
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import argparse
+import logging
+
+import apache_beam as beam
+from apache_beam.dataframe.convert import to_dataframe
+from apache_beam.options.pipeline_options import PipelineOptions
+
+
+def get_delay_at_top_airports(aa):
+  arr = aa.rename(columns={'arrival_airport': 'airport'}).airport.value_counts()
+  dep = aa.rename(columns={
+      'departure_airport': 'airport'
+  }).airport.value_counts()
+  total = arr + dep
+  # Note we keep all to include duplicates.
+  # This ensures the result is deterministic
+  top_airports = total.nlargest(10, keep='all')
+  return aa[aa['arrival_airport'].isin(top_airports.index.values)].mean()
+
+
+def input_date(date):
+  import datetime
+  parsed = datetime.datetime.strptime(date, '%Y-%m-%d')
+  if parsed > datetime.datetime(2012, 12, 31):
```

Comment:
Is there a minimum date as well?

Reply:
Good point, there is. Added another check for it.